Userland/Libraries/LibWeb/HTML/AbstractBrowsingContext.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Forward.h>
+#include <LibJS/Heap/Cell.h>
+#include <LibWeb/HTML/HistoryHandlingBehavior.h>
+#include <LibWeb/HTML/PolicyContainers.h>
+#include <LibWeb/HTML/WindowProxy.h>
+
+namespace Web::HTML {
+
+class AbstractBrowsingContext : public JS::Cell {
```

Comment:
Hmm I don't *love* the name `AbstractBrowsingContext`. I think it would be better to have:
`BrowsingContext` -> virtual base
`LocalBrowsingContext` - inherit from `BrowsingContext` for this WC process (what is currently `BrowsingContext`)
`RemoteBrowsingContext` - inherit from `BrowsingContext` for remote WC proccesses

...but that might be a huge pain to rename the existing `BrowsingContext` to `LocalBrowsingContext`, so I won't push too hard here. Maybe I'm just having Java flashbacks :smile:

Reply:
yes, I totally agree. maybe in another PR? :) I would prefer to try do move stuff with this approach of splitting BC into remote and local and then go for this large renaming :)